File: server/src/modules/auth/domain/repositories/auth.repository.ts
export type RegisterAuthInput = {
  email: string;
  password: string;
  confirmPassword: string;
  phone: string | null;
  fullName: string;
  avatarUrl?: string;
  acceptTerms: boolean;
};

export type RegisteredAuthUser = {
  id: string;
  email: string;
};

export abstract class AuthRepository {
  abstract register(input: RegisterAuthInput): Promise<RegisteredAuthUser>;
  abstract deleteUser(userId: string): Promise<void>;
  abstract login(email: string, password: string): Promise<{ token: string, userId: string }>;
  abstract findUserIdByEmail(email: string): Promise<string | null>;
  abstract updatePassword(userId: string, password: string): Promise<void>;
}

File: client/packages/domain/lib/src/auth/repositories/auth_repository.dart
import 'package:core/errors.dart';
import 'package:fpdart/fpdart.dart';

import '../../../auth.dart';

abstract interface class AuthRepository {
  Stream<UserEntity?> get onAuthStateChanged;

  Future<Either<Failure, void>> register({
    required String fullName,
    required String password,
    required String confirmPassword,
    required String email,
    required String phone,
    required bool acceptedTerms,
  });

  Future<Either<Failure, void>> logout();
}
